operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperator.java

```diff
@@ -0,0 +1,238 @@
+                && newVersion.getMetadata() != null) {
+            return !Objects.equals(oldVersion.getMetadata().getResourceVersion(), newVersion.getMetadata().getResourceVersion());
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Creates a resource with the name with the given desired state
+     * and completes the given future accordingly.
+     */
+    protected Future<ReconcileResult<T>> internalCreate(String name, T desired) {
+        try {
+            ReconcileResult<T> result = ReconcileResult.created(operation().withName(name).create(desired));
+            log.debug("{} {} has been created", resourceKind, name);
+            return Future.succeededFuture(result);
+        } catch (Exception e) {
+            log.error("Caught exception while creating {} {}", resourceKind, name, e);
+            return Future.failedFuture(e);
+        }
+    }
+
+    /**
+     * Synchronously gets the resource with the given {@code name}.
+     * @param name The name.
+     * @return The resource, or null if it doesn't exist.
+     */
+    public T get(String name) {
+        return operation().withName(name).get();
+    }
+
+    /**
+     * Asynchronously gets the resource with the given {@code name}.
+     * @param name The name.
+     * @return A Future for the result.
+     */
+    public Future<T> getAsync(String name) {
+        Future<T> result = Future.future();
+        vertx.createSharedWorkerExecutor("kubernetes-ops-tool").executeBlocking(
+            future -> {
+                T resource = get(name);
+                future.complete(resource);
+            }, true, result.completer()
+        );
+        return result;
+    }
+
+    /**
+     * Synchronously list the resources with the given {@code selector}.
+     * @param selector The selector.
+     * @return A list of matching resources.
+     */
+    @SuppressWarnings("unchecked")
+    public List<T> list(Labels selector) {
+        return listInAnyNamespace(selector);
+    }
+
+    protected List<T> listInAnyNamespace(Labels selector) {
+        FilterWatchListMultiDeletable<T, L, Boolean, Watch, Watcher<T>> operation = operation().inAnyNamespace();
+
+        if (selector != null) {
+            Map<String, String> labels = selector.toMap();
+            return operation.withLabels(labels)
+                    .list()
+                    .getItems();
+        } else {
+            return operation
+                    .list()
+                    .getItems();
+        }
+    }
+
+    /**
+     * Returns a future that completes when the resource identified by the given {@code name}
+     * is ready.
+     *
+     * @param name The resource name.
+     * @param pollIntervalMs The poll interval in milliseconds.
+     * @param timeoutMs The timeout, in milliseconds.
+     * @param predicate The predicate.
+     */
+    public Future<Void> waitFor(String name, long pollIntervalMs, final long timeoutMs, Predicate<String> predicate) {
```

Comment:
Why does this need to be a `public` method in this class? It seems to add very little on top of `Util.waitFor()`. Does it need to be overridden in a subclass? How many callers does it have? Can't you just inline it?